src/lib/operators/join_nested_loop.cpp

```diff
@@ -65,16 +65,15 @@ namespace opossum {
 
 JoinNestedLoop::JoinNestedLoop(const std::shared_ptr<const AbstractOperator>& left,
                                const std::shared_ptr<const AbstractOperator>& right, const JoinMode mode,
-                               const ColumnIDPair& column_ids, const PredicateCondition predicate_condition)
-    : AbstractJoinOperator(OperatorType::JoinNestedLoop, left, right, mode, column_ids, predicate_condition) {}
+                               const OperatorJoinPredicate& primary_predicate)
+    : AbstractJoinOperator(OperatorType::JoinNestedLoop, left, right, mode, primary_predicate, {}) {}
```

Comment:
@mrzzzrm: How much work would it be to add secondary predicates to the nested loop join (in a follow-up PR)? I see a value in having this as an easy-to-understand and complete reference join.

Reply:
I will extend the JNL into our ground-truth, omnipotent JoinOperator once this PR is merged. We need this in the context of #1542 anyway.